File: PruebaPacientes/PruebaPacientes/PruebaPacientes/AntecedenteEpidemiologicos2.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Modelo
{
    public class AntecedenteEpidemiologicos2
    {
       public int Id                                                    { get; set; }

       public bool VisitoCentroConCasosPositivosPrevioASintomas         { get; set; }
       public string NombreDelCentroVisitado                            { get; set; }
       public string CiudadDelCentroVisitado                            { get; set; }
       public DateTime FechaDeLaVisitaAlCentro                          { get; set; }
       
       public bool ContactoConAnimalesPrevioASintomas                   { get; set; }
       public TipoDeAnimalesConContacto TipoDeAnimalesConContacto       { get; set; }
       public string OtroTipoDeContactoConAnimales                      { get; set; }

      public bool ContactoConPersonasConIRPrevioASintomas               { get; set; }  //IR = Infeccion Respiratoria
      public TipoDeEntornoDeContacto TipoDeEntornoDeContacto            { get; set; }
      public string OtroTipoDeEntornoDeContacto                         { get; set; }

      public bool ContactoConCasosProbablesOConfirmadosPrevioASintomas  { get; set; } 
      public string ApellidoDelCaso                                     { get; set; }
      public string NombreDelCaso                                       { get; set; }
      public string DNIDelCaso                                          { get; set; }
      public string PaisDelCaso                                         { get; set; }
      public string AreaDelCaso                                         { get; set; }
      //public int FormularioParaAnalisisId { get; set; }
      //public FormularioParaAnalisis FormularioParaAnalisis { get; set; }


    }
}

File: Frontend/api/models/AntecedenteEpidemiologicos2.js
/**
 * AntecedenteEpidemoliogicos2.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {

    VisitoCentroConCasosPositivosPrevioASintomas:{
      type: 'boolean',
      defaultsTo: false
    },
    NombreDelCentroVisitado:{
      type: 'string'
    },
    CiudadDelCentroVisitado:{
      type: 'string'
    },
    FechaDeLaVisitaAlCentro:{
      type: 'string'
    },
    ContactoConAnimalesPrevioASintomas:{
      type: 'boolean',
      defaultsTo: false
    },
    TipoDeAnimalesConContacto:{ //Modelo Uno-Muchos
      model: 'TipoDeAnimalesConContacto'
    },
    OtroTipoDeContactoConAnimales:{
      type: 'string'
    },
    ContactoConPersonasConIRPrevioASintomas:{
      type: 'boolean',
      defaultsTo: false
    },
    TipoDeEntornoDeContacto:{ //Modelo
      model: 'TipoDeEntornoDeContacto'
    },
    OtroTipoDeEntornoDeContacto:{
      type: 'string'
    },
    ContactoConCasosProbablesOConfirmadosPrevioASintomas:{
      type: 'boolean',
      defaultsTo: false
    },
    ApellidoDelCaso:{
      type: 'string'
    },
    NombreDelCaso:{
      type: 'string'
    },
    DNIDelCaso:{
      type: 'string'
    },
    PaisDelCaso:{
      type: 'string'
    },
    AreaDelCaso:{
      type:'string'
    },
    //Relacion
    owner:{
      model:'FormularioParaAnalisis',
      unique: true
    }

  },

};

